### DISCUSSION TITLE
Work to be done for website

### DISCUSSION THREAD

**Discussion by KendallDoesCoding** [collab-community/journey-book#32] [General]
Following our discussion in #1 , creating a discussion for us to talk about what has to be done for the website.

cc: @Adarsh-jaiss @vikasganiga05

**Comment by vikasganiga05** [comment]
Hi folks, please share your thoughts on the website.

**Reply by KendallDoesCoding** [reply]
Hi Vikas,

Will share in a bit.

Thanks

**Reply by Adarsh-jaiss** [reply]
Kindly send the link :)

**Reply by Adarsh-jaiss** [reply]
https://unruffled-chandrasekhar-ffade4.netlify.app

This is displaying coming soon

**Reply by KendallDoesCoding** [reply]
> Kindly send the link :)

@Adarsh-jaiss, we haven't started work. I meant that will share thoughts in a bit.

**Comment by KendallDoesCoding** [comment]
Okay, so here are my thoughts and what needs to be done:

- We need to make a website link [Linkfree](https://linkfree.eddiehub.io) but here instead of the profile opening in browser if the button is pressed the file just downloads on the computer (I can try to do this, so this is assigned to me (@KendallDoesCoding)
- We had the option to always update the HTML when someone adds their journey, or to automatically update it using something similar to what Linkfree does, @Adarsh-jaiss decided to make it update automatically, to not give us extra work (assigned to @vikasganiga05)
     - @vikasganiga05 I would like to know in which language you can program the above, JS I hope can do this so I can work on the website accordingly.
- I have also asked @shourgamer2 who seems to be a good web-dev if he can help either me or @vikasganiga05 .
- I think a while ago @TechStudent11 said he could also help with the website, but he seems to be pretty busy these days so I don't want to bother him, in that case. 

I am free most of the time, please let me know when we can get to work on this @vikasganiga05 

@Adarsh-jaiss - Please invite @vikasganiga05 to be the repo collab
@vikasganiga05 - We will have to work on the website in the branch: [website](https://github.com/Adarsh-jaiss/journey-book/tree/website), the branch will deploy onto https://unruffled-chandrasekhar-ffade4.netlify.app/ for the time-being, when the website is made we will go into either buying the domain or getting a better domain for free, for that matter.

cc: @Adarsh-jaiss - Please confirm that only the above mentioned as to be done for the website.

Thanks,
Kendall Does Coding

**Reply by TechStudent10** [reply]
*Correction; I'm not always on GitHub, at least not as much as before.

**Reply by KendallDoesCoding** [reply]
Yeah, sorry I framed it badly

**Reply by Adarsh-jaiss** [reply]
@KendallDoesCoding yes everything is okay, if you could also add online reading about people journeys ( which means pepole can either read it online or can download people journeys)

**Reply by KendallDoesCoding** [reply]
> @KendallDoesCoding yes everything is okay, if you could also add online reading about people journeys ( which means pepole can either read it online or can download people journeys)

Ok, then instead of a text file, the .md file can be made into a PDF which can be read online as well, is that possible @vikasganiga05 ?

**Comment by vikasganiga05** [comment]
Hi, I found an excellent tool called [Docsify](https://docsify.js.org/). Using this we can convert MD to HTML. Also, it provides plugins to convert to PDF too. I don't know how to implement it, I need some time to go through docs or tutorial videos. Please take a look at it. :slightly_smiling_face:

**Reply by KendallDoesCoding** [reply]
Hi Vikas, yeah ik about Docsify, but I haven't gotten too deep into it, using the tool will make it much easier for us to do the website. Although just have some queries which I will leave below


- will docsify work with a Netlify site or will it only work with GitHub Pages?
- if the user opens the PDF in the browser, will they see a button to download the pdf to their computer/mobile, device.
- will we be able to add 2 buttons before the user opens the page, for either downloading on device or opening in browser.

**Reply by vikasganiga05** [reply]
Yes, it will work with Netlify too. Where do we store pdf files? If we store them on the project, then I found a gh-action. Which converts markdown to pdf, when we push changes to the specified branch(in our case main/website).

GitHub Action &rarr; [Markdown to PDF and HTML](https://github.com/marketplace/actions/markdown-to-pdf-and-html)

Docsify allows adding the `download` attribute to links. But we need to test this one.

**Reply by KendallDoesCoding** [reply]
> Yes, it will work with Netlify too

Oh cool.

> Where do we store pdf files?

We are just planning to store .md files in this repo but @Adarsh-jaiss wants a view of the .md file in the browser, hence I thought of the pdf idea, cause that will help a lot.

> If we store them on the project

As I stated above, we are only planning to store .md files in this repo, is it required for us to store pdf files too?

> Docsify allows adding the download attribute to links. But we need to test this one.

Ok, cool.

**Reply by vikasganiga05** [reply]
Yes, we need to convert .md to .pdf and store it in the repo. Because downloading the user journey in the website.

**Comment by vikasganiga05** [comment]
We are using Netlify. It can run some scripts(I mean build command) on code push to the repo. So, we can do like this :point_down: 

- Need to have two directories. one for `journeys` and one for `pdfs`(or `./asset/pdfs`).
- When the user adds and pushes his/her journey to the repo, Netlify will convert .md to pdf and store it in the `pdfs` directory.
- Need to find a good npm package that converts .md to .pdf. I have found one &rarr; [markdown-pdf](https://www.npmjs.com/package/markdown-pdf), but its last published was 3 years ago. If you know of any other packages please let me know.

For this to work properly, The journeys .md filenames should be unique. For Example `<your-github-username>.md`

**Reply by KendallDoesCoding** [reply]
> Need to have two directories. one for journeys and one for pdfs(or ./asset/pdfs).

Ok cool.

> When the user adds and pushes his/her journey to the repo, Netlify will convert .md to pdf and store it in the pdfs directory.

Will Netlify be able to do it or will we need a GitHub action?

> Need to find a good npm package that converts .md to .pdf. I have found one → [markdown-pdf](https://www.npmjs.com/package/markdown-pdf), but its last published was 3 years ago. If you know of any other packages please let me know.

Is this really required, if it is then I won't be able to work on this part because node, react, doesn't work on my laptop, computer.

> For this to work properly, The journeys .md filenames should be unique. For Example <your-github-username>.md

Yup ofc, we can inculde this in the instructions.

**Reply by vikasganiga05** [reply]
Sorry, I was busy with a hackathon. So, not started testing. Give me some time to work on this. Thank You :pray:

**Reply by KendallDoesCoding** [reply]
> Sorry, I was busy with a hackathon. So, not started testing. Give me some time to work on this. Thank You 🙏

Sure, take your time

**Comment by KendallDoesCoding** [comment]
@vikasganiga05 @Adarsh-jaiss  - Instead of making it so that, the files get downloaded on computer as .txt, let's just do a browser view of the file, as it will make our lives much easier, if people want to download the .md files, they can do it from this repo and I believe it's responsive in notepad, etc... 

If we do this, our life will become super easy as GitHub Pages can host .md files as a static website per page I believe, but I'm unsure, I need to test it further.

A few days ago, I made a .md file in my repo called [Best-Hosting-Service](https://github.com/KendallDoesCoding/Best-Hosting-Service) and using that .md file it generated a website with a theme of my choice: https://kendalldoescoding.github.io/Best-Hosting-Service... Although, as mentioned above, I am unsure if it only works for index.md file for GitHub pages or for all the .md files in the repo, so will test and let yall know if this works for yall.


Waiting for your reply.

**Reply by KendallDoesCoding** [reply]
I've created a repo to do testing, https://github.com/KendallDoesCoding/journey-book-test

**Reply by KendallDoesCoding** [reply]
@Adarsh-jaiss @vikasganiga05 Amazing, it WORKEDDD... I created 2 .md files and it generated 2 pages on the website.


First Page: https://kendalldoescoding.github.io/journey-book-test
Second Page: https://kendalldoescoding.github.io/journey-book-test/mrbeast

**Reply by KendallDoesCoding** [reply]
@Adarsh-jaiss - Please let me know if you like this idea, and you can get to setting up GitHub Pages.

@vikasganiga05 - We can then proceed on making a website with similar interface as Linkfree, with all the profiles listed, if possible. Please let me know

**Reply by KendallDoesCoding** [reply]
> @vikasganiga05 - We can then proceed on making a website with similar interface as Linkfree, with all the profiles listed, if possible. Please let me know

Actually, we can try to do this also within a .md file, @Adarsh-jaiss or @vikasganiga05 can you find a workflow that will take the names of .md files and display the GitHub pages preview link of that file and consolidate all those links with the name of the file in one .md file, if you get what I mean?

**Reply by vikasganiga05** [reply]
> > @vikasganiga05 - We can then proceed on making a website with similar interface as Linkfree, with all the profiles listed, if possible. Please let me know
> 
> Actually, we can try to do this also within a .md file, @Adarsh-jaiss or @vikasganiga05 can you find a workflow that will take the names of .md files and display the GitHub pages preview link of that file and consolidate all those links with the name of the file in one .md file, if you get what I mean?

No, I have not found any gh-action. We can solve this by adding a JSON file to the repo containing user data. Using this JSON file we can create links to their journey page(like a simple card with short details of the user). But, when a user adds his/her journey they need to make an entire on this file too.

**Reply by KendallDoesCoding** [reply]
Ok cool, will this work with GitHub pages tho @vikasganiga05

**Reply by KendallDoesCoding** [reply]
@vikasganiga05 Please clarify the following:

- Will this be able to be run locally
- @Adarsh-jaiss Is setting up the GitHub Pages site in the main branch so please get to work over there, if possible
Also, please let me know by when will this be ready to go/be done?

**Reply by KendallDoesCoding** [reply]
@vikasganiga05 Please let me know the above, also please let me know what the action is called.

**Reply by vikasganiga05** [reply]
> Ok cool, will this work with GitHub pages tho @vikasganiga05

No

**Reply by KendallDoesCoding** [reply]
Then how will it work @vikasganiga05 ?

**Comment by TechStudent10** [comment]
@KendallDoesCoding there is a problem with your idea; how do you upload to the repo? And what if someone puts something really bad? Then what? The repo get's deleted by GitHub? You really want to risk @Adarsh-jaiss account? You should upload the journey to a database and display it from there. You can use [Firebase](https://console.firebase.google.com) as a simple way to setup a database.

**Reply by KendallDoesCoding** [reply]
@TechStudent11 True, but for a normal user they make a PR, if they upload something bad, it won't be merged to the gh-branch, and the PR won't deploy on GitHub Pages.

For adding my journey, as I got access, I didn't make a PR and also it was solving issue #33 if I made my journey.

I'm just trying to make life easier for all of us, and it was just a idea that I shared, if everyones fine with the idea we can proceed, if not, then we'll do it the way we originally decided.

Thanks for sharing your thoughts

**Reply by TechStudent10** [reply]
> @TechStudent11 True, but for a normal user they make a PR, if they upload something bad, it won't be merged to the gh-branch, and the PR won't deploy on GitHub Pages.

How do you create a PR from the website? What if the user doesn't even have a GitHub account?

**Reply by KendallDoesCoding** [reply]
Nono, you need to have a GitHub account, there will not be a option to create your profile from the website, the website will only list all te profiles created.

**Reply by KendallDoesCoding** [reply]
Although, this is kinda a good idea.. @vikasganiga05 Would it be possible for users to add their profile on like a form page on the GitHub Pages website and their profile becomes made into a PR by a workflow, if possible. This is not required, but a IDEA.

**Reply by Adarsh-jaiss** [reply]
-- I totally a agree with @TechStudent11 , can't we use Firebase? (If possible)?

**Reply by Adarsh-jaiss** [reply]
Just asking, otherwise we should have to change our contributing guidelines and if possible we can find any GitHub action which can filter such words or lines which GitHub never tolerates

**Reply by KendallDoesCoding** [reply]
> -- I totally a agree with @TechStudent11 , can't we use Firebase? (If possible)?

I mean, we can yes. But that will take will/may be a lot of work, effort and I also think @TechStudent11 didn't understand my idea. We need people to make PR's, with those PR's it will be hosted via my idea, users can add to the repo from the website.

**Reply by KendallDoesCoding** [reply]
> Just asking, otherwise we should have to change our contributing guidelines and if possible we can find any GitHub action which can filter such words or lines which GitHub never tolerates

You too didn't understand my idea?

**Reply by KendallDoesCoding** [reply]
@Adarsh-jaiss @TechStudent11 @vikasganiga05 THIS IS MY IDEA.


As mentioned in README.md, users make a PR with their journey as a .md file in the journeys directory.
If the journey is fine, we merge it, if it is spam or inappropriate we don't merge it.
All the files merged will be hosted via the GitHub Pages website, as per my tests as attached in my previous messeges.

We will have to make a website then to just list all the journey's and their url which has been created by the .md file being in the repo.

As @TechStudent11 bought that up, even tho I wasn't thinking of it first, I thought we could do this -  Would it be possible for users to add their profile on like a form page on the GitHub Pages website and their profile becomes made into a PR by a workflow, if possible. This is not required, but a IDEA.

**Reply by Adarsh-jaiss** [reply]
Yes, I am little confused.. 
Because you are saying that we will not merge those pull request which contains that type of things which can harm our project as well as account. 

But what if, we merge it by mistake :) because afterall we are humans right! 
So, anyone can do mistake. So that's what I was worried about

**Reply by Adarsh-jaiss** [reply]
> @Adarsh-jaiss @TechStudent11 @vikasganiga05 THIS IS MY IDEA.
> 
> As mentioned in README.md, users make a PR with their journey as a .md file in the journeys directory. If the journey is fine, we merge it, if it is spam or inappropriate we don't merge it. All the files merged will be hosted via the GitHub Pages website, as per my tests as attached in my previous messeges.
> 
> We will have to make a website then to just list all the journey's and their url which has been created by the .md file being in the repo.
> 
> As @TechStudent11 bought that up, even tho I wasn't thinking of it first, I thought we could do this - Would it be possible for users to add their profile on like a form page on the GitHub Pages website and their profile becomes made into a PR by a workflow, if possible. This is not required, but a IDEA.

Now I got it.

**Reply by KendallDoesCoding** [reply]
Yes but, I personally and believe all open-source people should READ the files before merging into the repo? And also there are horrible websites hosted on GitHub pages, nothing will happen to your account, don't worry.

We just have to pay attention before merging, that's all.

**Reply by Adarsh-jaiss** [reply]
I would say, do whatever the project requires, but play safe guies 😄😄

**Reply by KendallDoesCoding** [reply]
> 

@TechStudent11 Thought that anyone even without using GitHub using the website can magically access the repo and commit BAD journey's using .md files, but that's not how it works. Users need to make their PR on github.

**Reply by KendallDoesCoding** [reply]
> I would say, do whatever the project requires, but play safe guies 😄😄

Yeah exactly.. @TechStudent11 misunderstood my idea, which is no worries but yeah

**Reply by Adarsh-jaiss** [reply]
> Yes but, I personally and believe all open-source people should READ the files before merging into the repo? And also there are horrible websites hosted on GitHub pages, nothing will happen to your account, don't worry.
> 
> We just have to pay attention before merging, that's all.

👍👍

**Reply by TechStudent10** [reply]
> > 
> 
> @TechStudent11 Thought that anyone even without using GitHub using the website can magically access the repo and commit BAD journey's using .md files, but that's not how it works. Users need to make their PR on github.

... Bruh

**Reply by TechStudent10** [reply]
@KendallDoesCoding I know that. **You** just confused us on what **your** idea is.

**Comment by KendallDoesCoding** [comment]
@Adarsh-jaiss ig you can delete the Netlify site as we will be using GitHub Pages now, you know how to set that up. Please setup a theme as well :)

**Reply by KendallDoesCoding** [reply]
I think the theme 'jekyll-theme-tactile' is brilliant for this project

**Reply by Adarsh-jaiss** [reply]
Okay I am gonna delete the Netlify, and please give me some time for setting up GitHub pages

**Reply by KendallDoesCoding** [reply]
Sure, please set it up as soon as possible though. Once GitHub Pages is setup, I can access the stuff and do required changes.

**Reply by Adarsh-jaiss** [reply]
Okay

**Reply by KendallDoesCoding** [reply]
Also, please set it up in the "main" branch.

**Reply by Adarsh-jaiss** [reply]
Main branch or main branch /journeys

**Reply by Adarsh-jaiss** [reply]
Just tell me this one because we only have to show . Md files by people not the entire repository  right!

**Reply by Adarsh-jaiss** [reply]
So, choose one and let me know, I will setup GitHub Pages

**Reply by KendallDoesCoding** [reply]
> Main branch or main branch /journeys

main branch as the other website also which @vikasganiga05 will make with .json has to be hosted, you may have to change the directory later tho, i'm not sure

**Reply by KendallDoesCoding** [reply]
the other website = the website to list all the previews to each journey

**Reply by KendallDoesCoding** [reply]
Let me know when the setup is done

**Reply by Adarsh-jaiss** [reply]
Okah

**Reply by KendallDoesCoding** [reply]
@Adarsh-jaiss Thanks, my journey is now hosted: https://adarsh-jaiss.github.io/journey-book/journeys/KendallDoesCoding

**Reply by Adarsh-jaiss** [reply]
Yeah 😄👍

**Comment by vikasganiga05** [comment]
Hi, @Adarsh-jaiss & @KendallDoesCoding , I created a test repo for testing the docsify features. Check it out :point_down: 
Your thoughts on this approach?

GitHub Repo &rarr; [neontuts/docsify-test](https://github.com/neontuts/docsify-test)
Live Preview &rarr; https://neontuts-docsify-test.netlify.app/

**Reply by Adarsh-jaiss** [reply]
It's looking great 😃... 
Can we add a search feature too?

**Reply by vikasganiga05** [reply]
Yes, there are plugins for that.

**Reply by KendallDoesCoding** [reply]
Hi Vikas,

Looking amazing, is there a database where you put the names to show though? How did it display those names.

**Reply by KendallDoesCoding** [reply]
Hey Vikas,

Only one issue. The text in the README doesn't look nice, is there another way of doing this by creating another file?

 {{ user.name }}
<script> new Vue({ el: '#usersList', data: function() { return { users: null, }; }, created: function() { fetch("Data.json") .then(res => res.json()) .then(data => (this.users = data)) .catch(err => console.log(err)); } }); </script>

**Reply by vikasganiga05** [reply]
> Hi Vikas,
> 
> Looking amazing, is there a database where you put the names to show though? How did it display those names.

It fetches user data from a JSON file called [Data.json](https://github.com/neontuts/docsify-test/blob/main/Data.json) on the repo's root.

**Reply by vikasganiga05** [reply]
> Hey Vikas,
> 
> Only one issue. The text in the README doesn't look nice, is there another way of doing this by creating another file?
> 
> {{ user.name }}
> <script> new Vue({ el: '#usersList', data: function() { return { users: null, }; }, created: function() { fetch("Data.json") .then(res => res.json()) .then(data => (this.users = data)) .catch(err => console.log(err)); } }); </script>

Yes, we can create `Index.md` file and add these scripts there. It's easy to configure.
Here is the docs &rarr; https://docsify.js.org/#/configuration?id=homepage

**Reply by KendallDoesCoding** [reply]
> > Hi Vikas,
> > Looking amazing, is there a database where you put the names to show though? How did it display those names.
> 
> It fetches user data from a JSON file called [Data.json](https://github.com/neontuts/docsify-test/blob/main/Data.json) on the repo's root.

Oh, but will this update automatically when users create their file?, yes right

**Reply by KendallDoesCoding** [reply]
> > Hey Vikas,
> > Only one issue. The text in the README doesn't look nice, is there another way of doing this by creating another file?
> > {{ user.name }}
> > <script> new Vue({ el: '#usersList', data: function() { return { users: null, }; }, created: function() { fetch("Data.json") .then(res => res.json()) .then(data => (this.users = data)) .catch(err => console.log(err)); } }); </script>
> 
> Yes, we can create `Index.md` file and add these scripts there. It's easy to configure. Here is the docs → https://docsify.js.org/#/configuration?id=homepage

Can you please do that :)

**Reply by vikasganiga05** [reply]
Okay, in the test repo right?

**Reply by KendallDoesCoding** [reply]
Yes

**Reply by vikasganiga05** [reply]
Hi @KendallDoesCoding , I changed the homepage of the website. Now the `README.md` file looks good. And sorry for the delay :slightly_smiling_face: 

#### Check it out :point_down: 
GitHub Repo → [neontuts/docsify-test](https://github.com/neontuts/docsify-test)
Live Preview → https://neontuts-docsify-test.netlify.app/

**Reply by vikasganiga05** [reply]
> It's looking great smiley... Can we add a search feature too?

I added the search plugin but it searches only the homepage not include the journeys dir.

**Comment by vikasganiga05** [comment]
I am very bad at design anyone can create a simple card for the home page to show all the user's cards. Try to use one color while designing the cards.
Here is the wireframe for that :point_down: 

![image](https://user-images.githubusercontent.com/74750414/175787355-77f21acf-d2ef-43d4-b8b2-fe688c2f97d6.png)

**Reply by Adarsh-jaiss** [reply]
You can use light blue color for making cards

**Reply by Adarsh-jaiss** [reply]
Or light green

**Reply by KendallDoesCoding** [reply]
> I am very bad at design anyone can create a simple card for the home page to show all the user's cards. Try to use one color while designing the cards. Here is the wireframe for that 👇
> 
> ![image](https://user-images.githubusercontent.com/74750414/175787355-77f21acf-d2ef-43d4-b8b2-fe688c2f97d6.png)

Hi Vikas,

Can you please make a issue on this instead of including it in this discussion, thanks.

**Reply by vikasganiga05** [reply]
Okay i will create a issue for that :)

**Reply by KendallDoesCoding** [reply]
Thanks!

**Reply by vikasganiga05** [reply]
> Thanks!

You're welcome

**Comment by KendallDoesCoding** [comment]
@cojored as stated by you, you want to help with this, can you please let me know, what languages you know to code :)

thanks, 
kendall

cc: @vikasganiga05 @Adarsh-jaiss

**Comment by KendallDoesCoding** [comment]
Thanks a ton, Vikas.

Please give me and Adarsh collab access to the repo.

**Reply by vikasganiga05** [reply]
Hi @KendallDoesCoding & @Adarsh-jaiss , I added both of you to the repo please check it out.

**Reply by KendallDoesCoding** [reply]
Thanks!

**Comment by KendallDoesCoding** [comment]
It's not visible to me.

**Reply by vikasganiga05** [reply]
The search input is on the top left corner of the website.

<img src="https://user-images.githubusercontent.com/74750414/176757402-fc615ef1-24d2-45f9-bbfe-2f8b010e53c2.png" width="400px" />

**Comment by Adarsh-jaiss** [comment]
guys , is there any progress on this project? @KendallDoesCoding @vikasganiga05

**Reply by vikasganiga05** [reply]
What are your thoughts @Adarsh-jaiss and @KendallDoesCoding on continuing with the current website or implementing Docsify as I did in the test repo &rarr; [docsify-test](https://github.com/neontuts/docsify-test)?

**Reply by KendallDoesCoding** [reply]
@vikasganiga05 Need some inputs from your end

- Will the docsify update automatically when a profile is made?
- The docsify interface looks much better then the site I generated.

**Reply by vikasganiga05** [reply]
> @vikasganiga05 Need some inputs from your end
> 
>     * Will the docsify update automatically when a profile is made?
> 
>     * The docsify interface looks much better then the site I generated.

Yes, It will update automatically.

**Reply by Adarsh-jaiss** [reply]
I think doscify is pretty good! 
Can we do one thing? 
Can we migrate this whole repository to our community repository?

**Reply by Adarsh-jaiss** [reply]
Because it will be more easy to attract users then, after migrating this whole project to our community repository I.e  @collab-community

**Reply by vikasganiga05** [reply]
Okay :+1:

**Reply by Adarsh-jaiss** [reply]
@vikasganiga05  what are your views on migrating this repository?

**Reply by vikasganiga05** [reply]
Yeah! I am okay with that, More people from the @collab-community will be interested in this project. After migrating this repo, no data will be lost right. like discussions etc.

**Reply by KendallDoesCoding** [reply]
What's the community repo

**Reply by Adarsh-jaiss** [reply]
community id: @collab-community, for migrating projects, I can create a repo ....
@KendallDoesCoding  if possible kindly transfer this repo to the community's Github , since we re going to announce about these projects very soon :) 

and if possible kindly complete the project as soon as possible :)

**Reply by KendallDoesCoding** [reply]
Hi Adarsh, Shouldn't it be a organization if it's a community and not a person github account?

YESS, it's possible to transfer repo very easily, if your the owner of the account more easy, if someone else owns the account, you will have to request them to accept the migration request.

**Reply by KendallDoesCoding** [reply]
@vikasganiga05 Please clear up the docsify website and do required changes and make a PR, I will then add some stuff the to website, etc.

**Reply by KendallDoesCoding** [reply]
@vikasganiga05 I am creating a seprate branch called "final" in the docsify-test repo, in which we will do the final code and then create a PR here.

**Reply by vikasganiga05** [reply]
Okay :+1:

**Reply by KendallDoesCoding** [reply]
> > @vikasganiga05 Need some inputs from your end
> > ```
> > * Will the docsify update automatically when a profile is made?
> > 
> > * The docsify interface looks much better then the site I generated.
> > ```
> 
> Yes, It will update automatically.

Hi, 

I did a test in commit [7248704](https://github.com/neontuts/docsify-test/commit/7248704db1f5f16aa64260ac83e1597d5774a3f8) and it didn't work automatically.

**Reply by KendallDoesCoding** [reply]
> > > @vikasganiga05 Need some inputs from your end
> > > ```
> > > * Will the docsify update automatically when a profile is made?
> > > 
> > > * The docsify interface looks much better then the site I generated.
> > > ```
> > 
> > 
> > Yes, It will update automatically.
> 
> Hi,
> 
> I did a test in commit [7248704](https://github.com/neontuts/docsify-test/commit/7248704db1f5f16aa64260ac83e1597d5774a3f8) and it didn't work automatically.

I think I messed something up on this by accident, sorry.

**Reply by vikasganiga05** [reply]
The user needs to add the file and make a new entry on the `Data.json` file to list on the home page.

**Reply by KendallDoesCoding** [reply]
Ooh, I thought a gh-action would do that for them but no worries!! We should list that in the sharing your experience page.

**Reply by vikasganiga05** [reply]
> Ooh, I thought a gh-action would do that for them but no worries!! We should list that in the sharing your experience page.

Okay, We can implement that in the future.

**Reply by Adarsh-jaiss** [reply]
> Hi Adarsh, Shouldn't it be a organization if it's a community and not a person github account?
> 
> YESS, it's possible to transfer repo very easily, if your the owner of the account more easy, if someone else owns the account, you will have to request them to accept the migration request.

I am the owner of that GitHub account. So it would be great if we move it to that account :) 
Tell me one thing, what will work perfectly 
1. Transfer the repository after completing the project
2. Transfer the repository now and we will do left work after that 
Please let me know :) @KendallDoesCoding

**Comment by Adarsh-jaiss** [comment]
![Screenshot_2022-07-12-19-01-01-952_com android chrome](https://user-images.githubusercontent.com/96974600/178502138-54be1c48-6a35-4259-97eb-6b1d2174726c.jpg)

**Reply by KendallDoesCoding** [reply]
Yup, I knew about this.

**Comment by KendallDoesCoding** [comment]
@vikasganiga05 @Adarsh-jaiss I've worked on the website quite a bit, and added different pages, example What is this project, Why we created this project, etc using text from this repos README.md, Although, I wanted to know from you @vikasganiga05 , if there is a way to set it that if text updated in README.md it gets updated within the page as well, because otherwise we will have to update text in the README.md as well as in the page, and that causes double work, even though it's not really a big issue.

So let me know.. Thanks

PREVIEW: [https://deploy-preview-2--neontuts-docsify-test.netlify.app/](https://deploy-preview-2--neontuts-docsify-test.netlify.app/)

**Reply by KendallDoesCoding** [reply]
Also, another query I had Vikas, we've added the side bar write, if possible when a user creates a journey, can the sidebar update and add the name of the person who created the journey, so that way all journeys easily can be accessed via a hyperlink.


cc: @vikasganiga05 @Adarsh-jaiss

**Reply by vikasganiga05** [reply]
Yes, it automatically updates the page once we merged the PR or directly changed the file.

**Reply by KendallDoesCoding** [reply]
> Yes, it automatically updates the page once we merged the PR or directly changed the file.

Nono, I wanted the page to show in this sidebar

![image](https://user-images.githubusercontent.com/84712013/178796407-b3d2cf57-e06e-455b-bc57-9db659ce1156.png)


Is it possible for the README.md to read the .json and add it to the sidebar, or something like that via a gh-action?

**Reply by vikasganiga05** [reply]
Yes, it is possible but we need to find a gh-action to do that.

**Reply by KendallDoesCoding** [reply]
Ok cool, I'm bad at finding gh-actions but will have a look around.

**Reply by vikasganiga05** [reply]
No worries we will find one.

**Reply by KendallDoesCoding** [reply]
> No worries we will find one.

👍

**Reply by KendallDoesCoding** [reply]
This is the best I could find: https://github.com/marketplace/actions/json2table

**Comment by KendallDoesCoding** [comment]
Yes, but in my eyes, a community should be a organization, You, Vikas, Me
can always be the owners of the organization, but it's up to you :)

The website is pretty much done, so you can transfer the repo now itself,
but it's up to you.

**Reply by Adarsh-jaiss** [reply]
We are the owners bro! 
That is our community. I am the founder of that community bro

**Reply by KendallDoesCoding** [reply]
Yes, but all the community's I've seen are GitHub organization, can you make it a GitHub organization? But it's up to you :)

Ex: EddieHubCommunity

**Reply by vikasganiga05** [reply]
> Yes, but all the community's I've seen are GitHub organization, can you make it a GitHub organization? But it's up to you :)
> 
> Ex: EddieHubCommunity

Your right, but they already made it has a GitHub account and started working on it.

**Comment by collab-community** [comment]
-- @KendallDoesCoding @vikasganiga05  guys, I have transferred the repository.
-- kindly check out,is everything is working perfectly or not :)

**Reply by KendallDoesCoding** [reply]
Yup, will do!

**Reply by KendallDoesCoding** [reply]
@collab-community Everything seems to be working fine, Although, Request you to please fork this repo onto your @Adarsh-jaiss account.

**Reply by KendallDoesCoding** [reply]
> @collab-community Everything seems to be working fine, Although, Request you to please fork this repo onto your @Adarsh-jaiss account.

Oops sorry, please fork it with another repo name tho, coz otherwise redirects wont work.

**Reply by KendallDoesCoding** [reply]
Fork it with like journey-book-fork or something

**Reply by Adarsh-jaiss** [reply]
yes i have forked it :)

**Reply by KendallDoesCoding** [reply]
Thanks a ton!

**Comment by KendallDoesCoding** [comment]
Yeah, so it's fine now!

**Reply by collab-community** [reply]
great :)

**Comment by KendallDoesCoding** [comment]
@vikasganiga05 Please confirm if the website is good to go and will make @collab-community (@Adarsh-jaiss ) review it.

What we require to do, after merging the website to this repo or before is:

- gh-action to update text (sections) from README.md to the individual page of the website (not so important)
- gh-action to hyperlink the name of the user from .json file and link their journey via a hyperlink in the sidebar
- Maybe add a navbar.

**Reply by KendallDoesCoding** [reply]
https://deploy-preview-2--neontuts-docsify-test.netlify.app/

**Reply by vikasganiga05** [reply]
For the user hyperlinks on the sidebar, we don't need gh-action. It can be done using the Vue.js as done in :point_down: 

![image](https://user-images.githubusercontent.com/74750414/179073925-b96aa3e2-de8f-4767-916e-61e5bef0ed50.png)

**Reply by vikasganiga05** [reply]
Sorry, I forgot about it :sweat_smile:

**Reply by vikasganiga05** [reply]
By the way, the website looks good to me.

**Reply by KendallDoesCoding** [reply]
> For the user hyperlinks on the sidebar, we don't need gh-action. It can be done using the Vue.js as done in 👇
> 
> ![image](https://user-images.githubusercontent.com/74750414/179073925-b96aa3e2-de8f-4767-916e-61e5bef0ed50.png)

Hi, Please work on this, if possible can it get done by tommrow?
